core/pkg/observability/peeker.go

```diff
@@ -37,18 +37,29 @@ func (p *Peeker) Peek(_ *http.Request, resp *http.Response) {
 	// If the status code is not a success code (2xx), we need to send the response to
 	// the user so they can see what happened.
 	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-		// We need to read the response body to send it to the user
-		buf, _ := io.ReadAll(resp.Body)
-
 		p.Lock()
+		defer p.Unlock()
```

Comment:
this is a risky pattern---`defer` only runs when the function exits, not when the `{}` block exits

just invert the `if` statement and put the lock/defer at the top level of the function